pandas/tests/indexing/test_indexing.py

```diff
@@ -717,6 +717,13 @@ def test_index_type_coercion(self):
                     idxr(s2)['0'] = 0
                     assert s2.index.is_object()
 
+    def test_empty_index_for_empty_dataframe(self):
+        index = pd.Index([], name='idx')
+        result = pd.DataFrame(columns=['A'], index=index)
+        result['A'] = []
+        expected = pd.DataFrame(columns=['A'], index=index)
```

Comment:
can you move this test to pandas/tests/frame/test_indexing.py, name something like

test_setitem_with_empty_list_like